### DISCUSSION TITLE
what does it mean this Resolve Retime clip [TimeEffect.1]

### DISCUSSION THREAD

**Discussion by KazukiOmata** [AcademySoftwareFoundation/OpenTimelineIO#1692] [Q&A]
Hello all.

rencently I have known resolve supports OTIO.
I exported otio file from DaVinci Resolve. 
This timeline has retime clip.
exported otio file has clip written TimeEffect.1 like below.

```
 "effects": [
    {
        "OTIO_SCHEMA": "TimeEffect.1",
        "metadata": {
            "Resolve_OTIO": {
                "Key Frames": [
                    [
                        0.0,
                        0.0,
                        false,
                        0.0,
                        0.0,
                        0.0,
                        0.0
                    ],
                    [
                        0.33366666666666669,
                        0.33366666666666669,
                        false,
                        0.0,
                        0.0,
                        0.0,
                        0.0
                    ],
                    [
                        9.857069444444443,
                        14.618770833333333,
                        false,
                        0.0,
                        0.0,
                        0.0,
                        0.0
                    ]
                ],
                "Speed Ratio": 1.4830747531734839
            }
        },
        "name": "",
        "effect_name": ""
    },



....
....
```

Does anyone know this "keyframes" meaning?
especially,  what theses array element do represent in this otio?

<img width="912" alt="image" src="https://github.com/AcademySoftwareFoundation/OpenTimelineIO/assets/67169225/afb91501-e56b-4f8b-aafd-d4410f5bac64">

Thankyou

**Comment by tweak-wtf** [comment]
hey kazuki! ohayou gozaimasu from berlin :)
i think we were writing on the BMD forums. let's connect on this one ;)

**Comment by apetrynet** [comment]
Hi @KazukiOmata
OTIO is designed in a way that all implementations may store custom metadata, preferably under a "vendor/application key" (`Resolve_OTIO` in this case).
For the most part this metadata only makes sense to the application that wrote it or tools that are aware of it. In this case the metadata under `Resolve_OTIO` is there so Resolve can round trip the time effect. The arrays most likely represent key frames that describe where in time the changes are applied.
Since OTIO itself has a very limited amount of effects ATM, this metadata approach is a good way to describe effects that make sense to Resolve.

I'm curious, does Resolve also provide a simplified time effect via the otio schema? Are you able paste the entire TimeEffect json object?
